File: src/main/java/handler/OpponentRanksHandler.java
package handler;

import api.ApiCaller;
import api.DataCollector;
import collect.misc.OpponentRanks;

import java.util.Map;

public class OpponentRanksHandler {
    private DataCollector dataCollector = new DataCollector(new ApiCaller());
    OpponentRanks opponentRanks = new OpponentRanks(dataCollector);

    public Map<String, Map<String, Integer>> handleRequest(Map<String, String> input) {
        String sport = input.get("sport");
        return opponentRanks.getOpponentRanks(sport);
    }
}

File: src/main/kotlin/handler/OpponentRanksHandler.kt
package handler

import collect.misc.OpponentRanks

class OpponentRanksHandler {
    fun handleRequest(input: Map<String, String>): Map<String, Map<String, Int>> {
        val sport = input.getValue("sport")
        return getOpponentRanksData(sport)
    }

    fun getOpponentRanksData(sport: String): Map<String, Map<String, Int>> {
        return OpponentRanks().getOpponentRanks(sport)
    }
}